### PR TITLE
Issue #101 Fix config set command

### LINKED ISSUE #77: Remove the unnecessary preflight checks for linux hosts
There are some preflight checks and setup which are not required:

1. default pool's availability and free space.
2. any others..

should be done after #63 is merged.

### LINKED ISSUE #87: Expose setup command for use
If there a reason to set `Hidden: true` for this command? if so, the current failue of a pre-flight does not mention the possibility to run setup.

https://github.com/code-ready/crc/blob/00b32714c6fa3fd8762258e656d50bd13e45e187/cmd/crc/cmd/setup.go#L19

### LINKED ISSUE #101: Config set is not working from the master
```console
 $ ./crc config -h | grep skip-check-libvirt-driver
 * skip-check-libvirt-driver

$ ./crc config set skip-check-libvirt-driver true
crc - Local OpenShift 4.0 clusters
Config property doesnot exist: skip-check-libvirt-driver
```

### LINKED ISSUE #105: setup command does not consider config properties
`crc setup` should skip the checks it performs if the check is configured to be skipped, Right now its not considering the config values.

https://github.com/code-ready/crc/blob/1e19504bf29e3238e8d8ee2d5bf9614347d4653b/pkg/crc/preflight/preflight_common.go#L29-L40

### PR DESCRIPTION
**Pull request by anjannath**
Fixes #101
Fixes #105 
Fixes #77 
Fixes #87

### CODE REVIEW

**Comment by anjannath** [comment]
To test this PR, try to set a config value, you can use `crc config help` to list out all the config properties. If you try to set a non existing property it should say that the property does not exist

Sample run:

```
[anjan@localhost]~/github.com/code-ready/crc% crc config set skip-check-crc-network true
crc - Local OpenShift 4.0 clusters
[anjan@localhost]~/github.com/code-ready/crc% crc config view                           
crc - Local OpenShift 4.0 clusters
- skip-check-crc-network             : true
- skip-check-default-pool            : true
[anjan@localhost]~/github.com/code-ready/crc% crc config set skip-check-crc-dnsmasq-file false
crc - Local OpenShift 4.0 clusters
[anjan@localhost]~/github.com/code-ready/crc% crc config view                                 
crc - Local OpenShift 4.0 clusters
- skip-check-crc-dnsmasq-file        : false
- skip-check-crc-network             : true
- skip-check-default-pool            : true
[anjan@localhost]~/github.com/code-ready/crc% crc config set skip-check-non-existing true
crc - Local OpenShift 4.0 clusters
Config property does not exist: skip-check-non-existing
```

File: Makefile [hunk 1]
```diff
@@ -72,4 +72,5 @@ test:
 .PHONY: clean ## Remove all build artifacts
 clean:
 	rm -rf $(BUILD_DIR)
+	rm $(GOPATH)/bin/crc
```
Comment (anjannath) [inline Makefile]: I need to remove this from time to time, so putting it in the makefile....

File: pkg/crc/machine/libvirt/constants.go [hunk 2]
```diff
@@ -6,10 +6,6 @@ const (
 	DefaultCacheMode = "default"
 	DefaultIOMode    = "threads"
 
-	// Static reesources
-	PoolName = "crc"
```
Comment (anjannath) [inline pkg/crc/machine/libvirt/constants.go]: These weren't used in this repo or the https://github.com/code-ready/machine-driver-libvirt repo

File: Makefile (hunk 1 above)
Comment (gbraad) [reply Makefile]: :-/

File: pkg/crc/machine/libvirt/constants.go (hunk 2 above)
Comment (gbraad) [reply pkg/crc/machine/libvirt/constants.go]: not used anymore...

File: Makefile [hunk 3]
```diff
@@ -72,4 +72,5 @@ test:
 .PHONY: clean ## Remove all build artifacts
 clean:
 	rm -rf $(BUILD_DIR)
+	rm -f $(GOPATH)/bin/crc
```
Comment (praveenkumar) [inline Makefile]: :+1:

**Pull request merged**